src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexReadOperation.cs

```diff
@@ -635,15 +626,16 @@ private static bool WillAlwaysIncludeInResults(IndexType indexType, FieldsToFetc
                             break;
                     }
                 }
-                
-                while (true)
+
+                // We don't need to do any processing for the query beyond counting if we are getting a count.
+                while (query.IsCountQuery == false)
```

Comment:
why `IsCountQuery==false`, is that changed in the loop?
I assume you want to say "if it's count only, don't do any work", but should move that to a variable to make it clearer

Reply:
The idea is to never enter into the loop. The cost of checking that is negligible, that's why I put the comment in there.